x/ccv/provider/keeper/proposal.go

```diff
@@ -189,6 +189,12 @@ func (k Keeper) StopConsumerChain(ctx sdk.Context, chainID string, closeChan boo
 		k.DeleteVscSendTimestampsForConsumer(ctx, chainID)
 	}
 
+	// delete consumer commission rate
```

Comment:
Good catch. We've to make sure the rewards have been distributed to validators before deleting `ConsumerRewardsAllocation`.
I will create an issue for it since it's not trivial.